### ISSUE TITLE
on_upstream_http_request hook not called for sources with type = federation

### ISSUE THREAD

**Issue by aarne** [the-guild-org/conductor#432]
config:
```yaml
sources:
    - type: federation
      id: whatever
      config:
          supergraph:
              file: supergraph.graphql

endpoints:
    - path: /graphql
      from: whatever
      plugins:
          - type: "vrl"
            config:
                on_downstream_http_request:
                    from: inline
                    content: |
                        log("on_downstream_http_request", level:"info")

                on_upstream_http_request:
                    from: inline
                    content: |
                        log("on_upstream_http_request", level:"info")

                on_downstream_http_response:
                    from: inline
                    content: |
                        log("on_downstream_http_response", level:"info")

                on_downstream_graphql_request:
                    from: inline
                    content: |
                        log("on_downstream_graphql_request", level:"info")

```

output:

```
2024-02-11T14:24:08.605823Z  INFO vrl::stdlib::log: on_downstream_http_request internal_log_rate_secs=1 vrl_position=0
2024-02-11T14:24:08.60589Z  INFO vrl::stdlib::log: on_downstream_graphql_request internal_log_rate_secs=1 vrl_position=0
2024-02-11T14:24:08.633743Z  INFO vrl::stdlib::log: on_downstream_http_response internal_log_rate_secs=1 vrl_position=0
```
Log for `on_upstream_http_request` is missing

**Comment by dotansimha** [comment]
Hi @aarne ! Thank you for reporting this issue. 
I'm working on a fix now :)

**Comment by dotansimha** [comment]
Fixed in `v0.0.6-alpha.0` @aarne :)
